### DISCUSSION TITLE
Is there a way to recognize equation in pdf?

### DISCUSSION THREAD

**Discussion by HeroadZ** [pymupdf/PyMuPDF#763] [Looking for help] [answered]
The equation has a special representation in pdf. Is it possible to ignore it when extracting text since it is meaningless?

I have checked the format of several equations.
```python
blocks = page.getText("dict", flags=0)["blocks"]
pprint(blocks)
```

It seems like that the font of some characters in an equation starts with "CMMI", and flags is 6 while for plain text it is 4. Is there any explanation for flags and font?

**Comment by JorjMcKie** [comment, marked as answer]
> Is there any explanation for flags and font?

Look in the documenttion [here](https://pymupdf.readthedocs.io/en/latest/textpage.html#dictionary-structure-of-extractdict-and-extractrawdict) for explaining `flags`.
Concerning fonts: the span dictionary just contains the fontname. If the font is an embedded subset, then (in PDF) it starts with 6 arbitrary upper case letters followed by "+", followed by the fontname from which the subset was created, e.g. `VIBWDS+NimbusSanL-Bold`.
Because `VIBWDS` is totally meaningless and only serves as a unique identifier for some technical reason, it is omitted and only `NimbusSanL-Bold` is contained in the dict.

**Comment by JorjMcKie** [comment]
I am not sure where "CMMI" came from in your question. Maybe purely arbitrary in the example you have been looking at.
To be more explicit with flags = 6 = 0B110, which corresponds to "italic, serifed".
Not even monospaced. So I would say, it is not very typical ...

**Reply by HeroadZ** [reply]
If I don't misunderstand you, you mean that there is no standard format for the equation in pdf, right? Pdf just save equations in normal characters in some common format like "italic", not using some special font like "Cambria Math" in ms word. In this situation, I have to recognize it in a new way.

**Reply by JorjMcKie** [reply]
Yes, exactly!
In PDF, text is just text. The PDF specification contains nothing to sub-divide different kinds of text. Equations are also text and be coded in any font, can be italic, or normal, mono-spaced of proportional, serifed or sans-serifed.
Also note that the equation symbol appears in program code listings a lot - ``PyMuPDF.pdf`` is full of such examples.

So I would say, that you have to develop your own way of recognizing equations ... and whatever you will develop, may not work with the next PDF example.

**Comment by JorjMcKie** [comment]
> The equation has a special representation in pdf.

I am not at all sure that this is true!

>  Is it possible to ignore it when extracting text since it is meaningless?

That would be your responsibility as a programmer. PyMuPDF cannot take on a position to judge the **_meaning_** of extracted text. This obviously is a matter of context - PyMuPDF is not in the semantics business.

**Reply by HeroadZ** [reply]
I see. Thank you very much. I think I need to rethink how to solve this question in a new way :D

**Comment by MasterYip** [comment]
I've found something useful by asking GPT here. Hope this is useful.
For example:
                        "size": 9.962599754333496,
                        "flags": 6,
                        "font": "CMMI10",
                        "color": 0,
                        "ascender": 0.75,
                        "descender": -0.25,
                        "text": " n",
                        "origin": [
                            423.5783996582031,
                            342.6888122558594
                        ],
                        "bbox": [
                            423.5783996582031,
                            335.21685791015625,
                            434.1545715332031,
                            345.1794738769531
                        ]
The font name "CMMI10" is a reference to a specific typeface in the Computer Modern font family. This font family was created by Donald Knuth and is commonly used in typesetting documents, especially with the LaTeX typesetting system.
In this context, the font name "CMMI10" indicates that the text "n" is rendered using the *Computer Modern Math Italic* font at size 10 points. The font is mathematically italic, which means it is specifically designed for mathematical symbols and expressions.
